### ISSUE TITLE
Add setter method for the View.projection_ field.

### ISSUE THREAD

**Issue by oxyeez** [openlayers/openlayers#15610]
**Is your feature request related to a problem? Please describe.**
For a project I need to display the map using the UTM/UPS projection system, and with this system, the projection have to be updated depending on where we are positioned on the map. 
So I need to update the ol/Projection given to my ol/View object each time I move on the map, but it's not possible because there is no setter for that field.
The solution I come with first was to recreate a View object with a new projection each time I needed to switch projection, but this added a lot of complexity in my code.

**Describe the solution you'd like**
To solve this, would it be possible to add a setter for the View.projection_ field, so that there is no need to recreate an instance of View each time I need to switch projection.

**Comment by ahocevar** [comment]
It would be possible to add that, but usually it is easier to do what you are currently doing - creating a new `View` and call `map.setView(newView)`. I don't see how this would be more complex than setting a projection, calculating the new center and zoom, and set the resulting properties on an existing view.

**Comment by mike-000** [comment]
It makes sense to create a new view as center and resolution must also change when the projection changes.  As long as the projections are defined using proj4 and their extents overlap the `onChangeProjection` function in https://openlayers.org/en/latest/examples/projection-and-scale.html should work for any combination of projections.

**Comment by oxyeez** [comment]
> It would be possible to add that, but usually it is easier to do what you are currently doing - creating a new `View` and call `map.setView(newView)`. I don't see how this would be more complex than setting a projection, calculating the new center and zoom, and set the resulting properties on an existing view.

Yeah I didn't explain much. The reason why it's adding complexity is because for this project I'm setting many listeners on the view for different purposes. So any time I'm replacing the old view by the new one I have to unset the listeners on the old one, and reset all of them for the new view. And this has to be done any time we change projection, so with the UTM/UPS system it can be very frequent (nearly every mouse drag).


> It makes sense to create a new view as center and resolution must also change when the projection changes. As long as the projections are defined using proj4 and their extents overlap the `onChangeProjection` function in https://openlayers.org/en/latest/examples/projection-and-scale.html should work for any combination of projections.

This is indeed the work from which one I started.
